contribs/gmf/src/directives/objecteditinggetwmsfeature.js

```diff
@@ -0,0 +1,159 @@
+goog.provide('gmf.ObjecteditinggetwmsfeatureController');
+goog.provide('gmf.objecteditinggetwmsfeatureDirective');
+
+goog.require('gmf');
+goog.require('gmf.ObjectEditingQuery');
+
+
+/**
+ * When activated, this directive registers clicks on an OL3 map and use the
+ * clicked coordinate to fetch a feature using the ObjectEditing query service.
+ * A feature returned is pushed to a collection.
+ *
+ * Example:
+ *
+ *     <gmf-objecteditinggetwmsfeature
+ *         gmf-objecteditinggetwmsfeature-active="ctrl.active"
+ *         gmf-objecteditinggetwmsfeature-features="ctrl.features"
+ *         gmf-objecteditinggetwmsfeature-layerinfo="ctrl.layerInfo"
+ *         gmf-objecteditinggetwmsfeature-map="::ctrl.map">
+ *     </gmf-objecteditinggetwmsfeature>
+ *
+ * @htmlAttribute {boolean} gmf-objecteditinggetwmsfeature-active Whether the
+ *     directive is active or not.
+ * @htmlAttribute {ol.Collection} gmf-objecteditinggetwmsfeature-features
+ *     The collection of features where to add those created by this directive.
+ * @htmlAttribute {gmf.ObjectEditingQuery.QueryableLayerInfo} gmf-objecteditinggetwmsfeature-layerinfo Queryable layer info.
+ * @htmlAttribute {ol.Map} gmf-objecteditinggetwmsfeature-map The map.
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ * @ngdoc directive
+ * @ngname gmfObjecteditinggetwmsfeature
+ */
+gmf.objecteditinggetwmsfeatureDirective = function() {
+  return {
+    controller: 'GmfObjecteditinggetwmsfeatureController',
+    scope: {
+      'active': '=gmfObjecteditinggetwmsfeatureActive',
+      'features': '<gmfObjecteditinggetwmsfeatureFeatures',
+      'layerInfo': '=gmfObjecteditinggetwmsfeatureLayerinfo',
+      'map': '<gmfObjecteditinggetwmsfeatureMap'
+    },
+    bindToController: true,
+    controllerAs: 'gwfCtrl'
+  };
+};
+
+gmf.module.directive(
+  'gmfObjecteditinggetwmsfeature',
+  gmf.objecteditinggetwmsfeatureDirective);
+
+
+/**
+ * @param {!angular.Scope} $scope Scope.
+ * @param {gmf.ObjectEditingQuery} gmfObjectEditingQuery GMF ObjectEditing
+ *     query service.
+ * @constructor
+ * @ngInject
+ * @ngdoc controller
+ * @ngname GmfObjecteditinggetwmsfeatureController
+ */
+gmf.ObjecteditinggetwmsfeatureController = function($scope,
+    gmfObjectEditingQuery) {
+
+  // Scope properties
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.active;
+
+  $scope.$watch(
+    function() {
+      return this.active;
+    }.bind(this),
+    this.handleActiveChange_.bind(this)
```

Comment:
I see.  Well, this would require a lot more than just this change (I think...)  I haven't checked but every time I implemented such watchers, I didn't take care of doing this.   I suggest we leave this as-is and if this really becomes a problem, then we could do a PR in which we would attempt to fix all watchers in all source files, not just those in this PR.  What do you think?